external-testing/README.md

```diff
@@ -0,0 +1,65 @@
+# External Testing
+
+Let's talk about testing process against external resources. What does that
+mean? Well, talking to GitHub API, COPR API, running `fedpkg`, cloning
+repositories...
+
+
+## Use cases
+
+* Interact with an external API (such as GitHub, COPR, Pagure, FAS, k8s)
+  * Authenticated vs. unauthenticated calls.
+* Clone a git repository.
+* Construct a git repository.
+* Run a command to perform an action.
+  * Changes happening only locally.
+  * RO access to a remote service.
+  * R/W access to a remote service.
+* Combination of multiple use cases into a single test case.
+
+
+## Solutions
+
+* flexmock
+
+* pytest fixtures
+
+* requre
+
+* VCRpy
+
+* Betamax
+
+
+## VCRpy vs. requre
+
+| A feature | VCRpy | requre |
+|-----------|-------|--------|
+| Record & replay for python requests | ✓ | ✓ |
+| Match on call stack | ✘ | ✓ |
+| Match on URL & request type | ✓ | ✘ |
```

Comment:
it is possible also in requre (but it is harder to use, but could be improved to store url as well), because requre is more generic than just to `request` module. 
In fact `requre` allows you to store any parameter of function as key (url) as well, but we have to ensure to remove sensitive data like login and password from url